composer/optim/scheduler.py

```diff
@@ -68,137 +64,110 @@ def _convert_time(time: Union[str, Time[int], Time[float]], state: State, ssr: f
-        ssr (float): The scale schedule ratio. In general, the learning rate computed by this
-            scheduler at time :math:`t` with an SSR of 1.0 should be the same as that computed by
-            this scheduler at time :math:`t \times s` with an SSR of :math:`s`. Default = ``1.0``.
         start_factor (float): Start factor. Default = ``1.0``.
         end_factor (float): End factor. Default = ``0.0``.
         total_time (str or Time): Total time. Default = ``'1dur'``.
     """
 
-    total_time = _convert_time(total_time, state, ssr=ssr)
-    current_time = state.timer.get(total_time.unit)
-    frac_of_total = min(1.0, (current_time / total_time).value)
+    def __init__(self, start_factor: float = 1.0, end_factor: float = 0.0, total_time: Union[str, Time] = '1dur'):
+        self.start_factor = start_factor
+        self.end_factor = end_factor
+        self.total_time = Time.from_timestring(total_time) if isinstance(total_time, str) else total_time
 
-    current_factor = start_factor + frac_of_total * (end_factor - start_factor)
+    def __call__(self, state: State, ssr: float = 1.0):
+        total_time = _convert_time(self.total_time, state, ssr=ssr)
+        current_time = state.timer.get(total_time.unit)
+        frac_of_total = min(1.0, (current_time / total_time).value)
 
-    return current_factor
+        current_factor = self.start_factor + frac_of_total * (self.end_factor - self.start_factor)
 
+        return current_factor
+
+
+class ExponentialScheduler:
+    """Decays the learning rate exponentially.
 
-def exponential_scheduler(state: State, *, ssr: float = 1.0, gamma: float) -> float:
-    r"""Decays the learning rate exponentially.
-    
     Args:
-        state (State): The current Composer Trainer state.
-        ssr (float): The scale schedule ratio. In general, the learning rate computed by this
-            scheduler at time :math:`t` with an SSR of 1.0 should be the same as that computed by
-            this scheduler at time :math:`t \times s` with an SSR of :math:`s`. Default = ``1.0``.
         gamma (float): Gamma.
+        decay_period (str or Time, optional): The period over which to decay the learning rate by gamma.
+            The learning rate is approximately calculated as ``lr = gamma ** (current_time / decay_period)``.
+            (default: ``'1ep'``)
     """
 
-    current_time = state.timer.epoch
+    def __init__(self, gamma: float, decay_period: Union[str, Time] = '1ep'):
+        self.gamma = gamma
+        self.decay_period = Time.from_timestring(decay_period) if isinstance(decay_period, str) else decay_period
 
-    return gamma**(current_time.value / ssr)
+    def __call__(self, state: State, ssr: float = 1.0):
+        decay_period = _convert_time(self.decay_period, state, ssr)
+        current_time_in_decay_units = state.timer.get(decay_period.unit) / ssr
 
+        return self.gamma**float(current_time_in_decay_units / decay_period)
```

Comment:
While equivalent, I think people will naturally look for it by name rather than know they need to configure a StepScheduler with step=1, so in favor of keeping it and removing that extra step.

Reply:
Actually, I guess this adjusts the learning rate continuously whereas `StepScheduler` adjusts it discretely, and both behaviors are probably desirable in their own ways.